include/volume.h

```diff
@@ -0,0 +1,272 @@
+// VolEsti
+
+// Copyright (c) 2012-2017 Vissarion Fisikopoulos
+
+// VolEsti is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// VolEsti is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+//
+// See the file COPYING.LESSER for the text of the GNU Lesser General
+// Public License.  If you did not receive this file along with HeaDDaCHe,
+// see <http://www.gnu.org/licenses/>.
+
+#ifndef VOLUME_H
+#define VOLUME_H
+
+#include <iterator>
+#include <fstream>
+#include <vector>
+#include <list>
+#include <algorithm>
+#include <math.h>
+#include "cartesian_geom/cartesian_kernel.h"
+#include "boost/random.hpp"
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/uniform_real_distribution.hpp>
+
+
+typedef double                      NT;
+typedef Cartesian<NT> 	      Kernel; 
+typedef Kernel::Point								Point;
+typedef boost::mt19937 RNGType; ///< mersenne twister generator
+
+
+//structs with variables and random generators
+struct vars{
+public:
+    vars( int m,
+          int n,
+          int walk_steps,
+          int n_threads,
+          const double err,
+          const double err_opt,
+          const int lw,
+          double up,
+          const int L,
+          RNGType &rng,
+          //std::uniform_real_distribution<NT> urdist,
+          //std::uniform_real_distribution<NT> urdist1,
+          boost::random::uniform_real_distribution<>(urdist),
+          boost::random::uniform_real_distribution<> urdist1,
+          bool verbose,
+          bool rand_only,
+          bool round,
+          bool NN,
+          bool birk,
+          bool coordinate
+          ) :
+        m(m), n(n), walk_steps(walk_steps), n_threads(n_threads), err(err), err_opt(err_opt),
+        lw(lw), up(up), L(L), rng(rng),
+        urdist(urdist), urdist1(urdist1) , verbose(verbose), rand_only(rand_only), round(round),
+        NN(NN),birk(birk),coordinate(coordinate){};
+
+    int m;
+    int n;
+    int walk_steps;
+    int n_threads;
+    const double err;
+    const double err_opt;
+    const int lw;
+    double up;
+    const int L;
+    RNGType &rng;
+    //std::uniform_real_distribution<NT> urdist;
+    //std::uniform_real_distribution<NT> urdist1;
+    boost::random::uniform_real_distribution<>(urdist);
+    boost::random::uniform_real_distribution<> urdist1;
+    bool verbose;
+    bool rand_only;
+    bool round;
+    bool NN;
+    bool birk;
+    bool coordinate;
+};
+
+
+#include "../external/LPsolve_src/run_headers/solve_lp.h"
```

Comment:
avoid relative position in #include i.e. ```"../../path"```
instead use ```"LPsolve/..."``` and in ```CMakeLists.txt``` add ```  include_directories (BEFORE ../external)```

Reply:
linking improved